### PR TITLE
Fix tables display

### LINKED ISSUE #58134: Blank right side border in table generated by rustdoc in Firefox
**Edit**: Add test code in metadata.

---

It is ugly in my crate (not public yet), see below image displayed in Firefox:

image 

but in Chromium:

image

At least in Chromium, the right side border of the table is displayed.

I tested in both 1.31.0, 1.32.0 and 1.34.0-nightly (852701ad6 2019-02-01), those have the same effect. I didn't test previous versions because I'm using Rust 2018.

## Metadata

<details>

<summary>OS and software info</summary>

OS: Xubuntu 18.04.1
Browser: Firefox 65.0 (All extensions are disabled)

### Rustdoc:

```command line
% rustdoc -Vv
rustdoc 1.32.0 (9fda7c223 2019-01-16)
binary: rustdoc
commit-hash: 9fda7c2237db910e41d6a712e9a2139b352e558b
commit-date: 2019-01-16
host: x86_64-unknown-linux-gnu
release: 1.32.0
LLVM version: 8.0
```

</details>
<details>
<summary>Build with below code to test locally with `rustdoc`</summary>

```rust
/// Tone (thanh điệu), which also called "accent", lists all possible
/// tones in Vietnamese.
///
/// These tones are:
///
/// Tone        | Example
/// ------------|--------
/// None        | a
/// Grave       | à
/// Acute       | á
/// Hook above  | ả
/// Tilde       | ã
/// Dot below   | ạ
#[derive(Debug, PartialEq)]
pub enum Tone {
    // Thanh bằng hoặc không dấu
    None = 0,
    // Thanh huyền
    Grave,
    // Thanh sắc
    Acute,
    // Thanh hỏi
    HookAbove,
    // Thanh ngã
    Tilde,
    // Thanh nặng
    DotBelow,
}
```

</details>

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Fixes #58134.

cc @lzutao

r? @QuietMisdreavus

### CODE REVIEW

**Comment by lzutao** [comment]
If the fix is in `dark.css` and `light.css`, I downloaded these patched files and placed it in `target/doc`.
The I disabled the browser cache, nothing has changed on my side.

**Comment by GuillaumeGomez** [comment]
Strange, I can't get to reproduce your bug with those changes... Also, when you're switching theme, if overwrites the css, so maybe try to build docs locally?

**Comment by lzutao** [comment]
I [zipped and uploaded the doc](https://github.com/rust-lang/rust/files/2855622/ff.zip). You could try to view it in Firefox. One table is in `gokien::tone::Tone` enum.

Hope that help!

**Comment by GuillaumeGomez** [comment]
Thanks a lot! Let's check what's going on. :3

**Comment by GuillaumeGomez** [comment]
First results: chrome works fine so maybe a firefox bug. Trying to find a workaround.

**Comment by GuillaumeGomez** [comment]
@lzutao Try again.

**Comment by lzutao** [comment]
@GuillaumeGomez Before
image

After
image

The settings button is in the search box now.

**Comment by GuillaumeGomez** [comment]
Try again bis.

**Comment by lzutao** [comment]
Seems to fix the issue. But the font SourceSerifPro-Regular cannot be used, current font is Times New Roman. In the pass I could bypass this error by going to the settings page then jump to crate doc on the left side menu.
```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at file:///stor/github.com/lzutao/rust/gokien/target/doc/SourceSerifPro-Regular.ttf.woff. (Reason: CORS request not http).[Learn More] 
```
image

**Comment by lzutao** [comment]
@GuillaumeGomez Those Traits Implementations section is disappear now. :cry: 
image

**Comment by lzutao** [comment]
Turn off `display: none` seems to display those Implementations again.
image

**Comment by GuillaumeGomez** [comment]
> Seems to fix the issue. But the font SourceSerifPro-Regular cannot be used, current font is Times New Roman. In the pass I could bypass this error by going to the settings page then jump to crate doc on the left side menu.

That is another issue. I'm thinking about an invalid path. To be determined...

> @GuillaumeGomez Those Traits Implementations section is disappear now. cry

That one, however, is very weird...

**Comment by GuillaumeGomez** [comment]
Can you update your rustdoc version and regenerate the doc **then** update the CSS and tell me if everything's fine.

**Comment by GuillaumeGomez** [comment]
(and please use nightly rustdoc for the current test)

**Comment by lzutao** [comment]
@GuillaumeGomez 
Without the patched, latest nightly `rustdoc` seems fix the missing side border issue. However, the missing cell issue stills exists.
After your patch
image

**Comment by GuillaumeGomez** [comment]
Missing cell is "expected" so I guess it's fine.

ping @QuietMisdreavus

File: src/librustdoc/html/static/themes/light.css [hunk 1]
```diff
@@ -67,21 +67,11 @@ pre {
...
-.docblock table td {
-	border-top-color: #ddd;
 	border-bottom-color: #ddd;
 }
 
-.docblock table th {
-	border-top-color: #ddd;
-	border-bottom-color: #ddd;
+.docblock table, .docblock table td, .docblock table th {
+	border-color: #000;
```
Comment (QuietMisdreavus) [inline src/librustdoc/html/static/themes/light.css]: This color in the light theme got changed from `#ddd` to `#000`. Was that intentional?

Comment (GuillaumeGomez) [reply src/librustdoc/html/static/themes/light.css]: No it wasn't, good catch!

**Comment by GuillaumeGomez** [comment]
Updated.!

**Review by QuietMisdreavus — APPROVED** [review]
Thanks!

**Pull request merged**